netlogo-core/src/main/api/LabProtocol.scala

```diff
@@ -23,14 +24,29 @@ case class LabProtocol(name: String,
 
   def elements: SettingsIterator = {
     def combinations(sets: List[ValueSet]): SettingsIterator =
-      sets match {
-        case Nil => Iterator(Nil)
-        case set::sets =>
-          set.iterator.flatMap(v =>
-            combinations(sets).map(m =>
-              (set.variableName,v) :: m))
-      }
-    combinations(valueSets)
+      useDefaultRunOrder match {
+      case true => 
+        sets match {
+          case Nil => Iterator(Nil)
+          case set::sets =>
+            set.iterator.flatMap(v =>
+              combinations(sets).map(m =>
+                (set.variableName,v) :: m))
+        }
+      case _ => 
+        sets match {
+          case Nil => Iterator(Nil)
+          case set::sets =>
+            set.iterator.flatMap(v =>
```

Comment:
Notice that line 30-32 and 38-40 are identical. Can you come up with a way to reduce this duplication?